mlflow/utils/databricks_utils.py

```diff
@@ -288,16 +288,13 @@ def is_databricks_connect(spark=None):
 
     if is_in_databricks_serverless_runtime() or is_in_databricks_shared_cluster_runtime():
         return True
-    try:
-        if spark is None:
-            spark = _get_active_spark_session()
-        # TODO: Remove the `spark.client._builder` attribute usage once
-        #  Spark-connect has public attribute for this information.
-        return is_spark_connect_mode() and any(
-            k == "x-databricks-cluster-id" for k, v in spark.client._builder.metadata()
-        )
-    except Exception:
-        return False
+
+    if hasattr(spark.client, "metadata"):
```

Comment:
Do we need check `hasattr(spark, "client")`? It's possible `spark` is None

Reply:
no.

now code is like
```
    if spark is None:
        spark = _get_active_spark_session()

    if not is_spark_connect_mode():
        return False

    if hasattr(spark.client, "metadata"):
```

if it is sparkconnect, then spark must have client attribute

Reply:
Maybe check if spark is None, if so return False :)

Reply:
```
def is_spark_connect_mode():
    try:
        from pyspark.sql.utils import is_remote
    except ImportError:
        return False
    return is_remote()

def is_remote() -> bool:
    """
    Returns if the current running environment is for Spark Connect.
    """
    return "SPARK_CONNECT_MODE_ENABLED" in os.environ
```
This doesn't mean spark is not None right?
Currently `if not is_spark_connect_mode(): return False` check is at the top

Reply:
oh :) you are right :)